### PR TITLE
Introduce Chip23

### PR DESCRIPTION
**Pull request by noakesmichael**
Introduce Chip 23 to provide supporting material for the new initializer proposal

### CODE REVIEW

**Comment by bradcray** [comment]
I've added a link to this CHIP in issue #8283.

File: doc/rst/developer/chips/23.rst [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+intent is avoid passing an object with uninitialized fields, and to
+avoid unpredictable/confusing interactions between components of the
+distributed implementation for the overall initializer.
+
+One choice is to allow derived classes to override one of both
+of init() and postInit() without any further extensions.  Additionally
+Derived.postInit() would not explictly invoke Parent.postInit(); the
+implementation would arrange to work "as if" from base class towards
+most derived class to ensure that a derived class can "own"
+the final value of every visible field.
```
Comment (cassella) [inline doc/rst/developer/chips/23.rst]: I'm having a hard time parsing the last part of this sentence -- to work as if *what* from base class towards most derived?  (Though I gather it means the ``postInit()``s are called in that order.)

File: doc/rst/developer/chips/23.rst [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+
+        <a user defined initializer>
+      }
+
+      var myR : MyR( ... );
+
+
+   and what is the signature for a user-defined initializer that
+   will satisfy the variable declaration?  Should the answer be
+   different if this were a class instead of a record?
```
Comment (cassella) [inline doc/rst/developer/chips/23.rst]: I don't understand what either of the questions in this section are getting at.  If it's clear to everyone on the team, feel free to ignore me.

File: doc/rst/developer/chips/23.rst [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+
+
+Initialization vs. assignment
++++++++++++++++++++++++++++++
+
+It is important to distinguish between initialization and assignment
+for fields with record type.  Mutable objects must be initialized once
+and may be assigned any number of times after that.  Failing to
+support this requirement correctly may have consequences for
+correctness or performance. Consider the following contrived example
```
Comment (cassella) [inline doc/rst/developer/chips/23.rst]: Ditto here.  I don't follow what bearing this has on the chip, or a difference relative to or wrt what's in master.   But ditto also the part about ignoring me.

**Pull request merged**